lib/Service/ContactsIntegration.php

```diff
@@ -105,14 +105,14 @@ public function getMatchingRecipient(string $term): array {
 	 * @return false|null|string
 	 */
 	public function getPhoto(string $email) {
-		$result = $this->contactsManager->search($email, ['EMAIL']);
-		foreach ($result as $contact) {
-			if (!isset($contact['PHOTO']) || empty($contact['PHOTO'])) {
-				continue;
+		$result = $this->contactsManager->search($email, ['EMAIL'], ['strict_search' => true]);
+		if (count($result) > 0) {
+			foreach ($result as $contact) {
+				if (isset($contact['PHOTO'])) {
+					return $this->getPhotoUri($contact['PHOTO']);
+				}
 			}
```

Comment:
I added this, because while testing note that if you create 2 identical contacts and the first one without a photo, and the second one with a photo, then the avatar is not loaded. 

I would still like to be able to find an avatar if there are duplicates of contacts and show it to the user. what do you think?

Reply:
@ChristophWurst

Reply:
The `if` is unecessary. Remove it. If there are no results the loop won't loop and `null` is returned.

The logic was and is that results with out a PHOTO are skipped. I don't see how you are changing the logic.

Reply:
Previously, a photo of the first element of the array ot null was return. 

thus, if there are >1 elements in the array and there is a photo in the second (third... etc.... ) element, we will still get null. 

I changed it, added a loop in which we go through all the elements exactly and return a photo if there is one, and null if no photo is found after iterating through the array

Reply:
I don't understand anything, for some reason I have an old version of the file. everything is already correct in the new MAIN )

Reply:
I believe this was changed in one of your other PRs :)

Reply:
Exactly 😊💯
The confusion arose due to the fact that I split one PR into 3

Reply:
[d05f6a3](https://github.com/nextcloud/mail/pull/6714/commits/d05f6a3442f827b0962fee5f8b84cbc234ee2832)